### PR TITLE
All changes to funky except persistence

### PR DESCRIPTION
**Pull request by asiegel-jt**

### CODE REVIEW

**Review by jumpsiegel — DISMISSED** [review]
For me, this is perfect but lets still have @kbowers-jump give a nod...

**Comment by mmcgee-jump** [comment]
Can we try keep the main history a little cleaner than a bunch of "funky-update: copy from private" commits?

**Comment by asiegel-jt** [comment]
> Can we try keep the main history a little cleaner than a bunch of "funky-update: copy from private" commits?

I don't recall the magic for squashing commits

File: src/funk/fd_funk_val.h [hunk 1]
```diff
@@ -40,8 +39,8 @@ fd_funk_val_max( fd_funk_rec_t const * rec ) { /* Assumes pointer in caller's ad
    NULL and vice versa.  Assumes no concurrent operations on rec. */
 
 FD_FN_PURE static inline void *         /* Lifetime is the lesser of rec or the value size is modified */
-fd_funk_val( fd_funk_rec_t *   rec,     /* Assumes pointer in caller's address space to a live funk record */
-             fd_wksp_t const * wksp ) { /* ==fd_funk_wksp( funk ) where funk is a current local join */
+fd_funk_val( fd_funk_rec_t const * rec,     /* Assumes pointer in caller's address space to a live funk record */
```
Comment (kbowers-jump) [inline src/funk/fd_funk_val.h:42]: This probably should return a `void const *` if it takes a `fd_funk_rec_t const *`.  Or maybe should have two versions for const correctness purposes.

File: src/funk/test_funk_val.c [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 #include "test_funk_common.h"
 
-FD_STATIC_ASSERT( FD_FUNK_REC_VAL_MAX==(10UL<<20), unit_test );
+FD_STATIC_ASSERT( FD_FUNK_REC_VAL_MAX==UINT_MAX, unit_test );
```
[the remark this answers is not in the document]
Comment (asiegel-jt) [reply src/funk/test_funk_val.c:7]: We need to store blocks which can be 128MB

File: src/funk/fd_funk_val.h (hunk 1 above)
Comment (asiegel-jt) [reply src/funk/fd_funk_val.h:42]: there are 2 versions. see fd_funk_val_const

File: src/funk/fd_funk_rec.c [hunk 3]
```diff
@@ -188,9 +188,61 @@ fd_funk_rec_modify( fd_funk_t *           funk,
...
+      txn = txn_map + txn_idx;
+      fd_funk_xid_key_pair_init( pair, fd_funk_txn_xid( txn ), rec->pair.key );
+    }
+
+    fd_funk_rec_t const * rec2 = fd_funk_rec_map_query( rec_map, pair, NULL );
+    if ( rec2 ) {
+      if ( rec->val_sz != rec2->val_sz )
+        return 1;
+      void * val2 = fd_funk_val( rec2, wksp );
+      return memcmp(val, val2, rec->val_sz) != 0;
```
[the remark this answers is not in the document]
Comment (asiegel-jt) [reply src/funk/fd_funk_rec.c:240]: The account API which is the next layer up has hashes of accounts.

File: src/funk/fd_funk_rec.h [hunk 4]
```diff
@@ -460,6 +467,8 @@ fd_funk_rec_write_prepare( fd_funk_t *               funk,
                            fd_funk_txn_t *           txn,
                            fd_funk_rec_key_t const * key,
                            ulong                     min_val_size,
+                           int                       do_create,
+                           fd_funk_rec_t const *     irec,
```
[the remark this answers is not in the document]
Comment (asiegel-jt) [reply src/funk/fd_funk_rec.h:481]: done

File: src/funk/fd_funk_txn.c [hunk 5]
```diff
@@ -456,10 +460,27 @@ fd_funk_txn_cancel_children( fd_funk_t *     funk,
...
+  ulong result = 0;
+  do {
+    ulong idx = fd_funk_txn_idx( funk->child_tail_cidx );
+    if ( idx==FD_FUNK_TXN_IDX_NULL )
+      break;
+    result += fd_funk_txn_cancel( funk, map+idx, verbose );
+  } while (1);
+  return result;
+}
+
```
[the remark this answers is not in the document]
Comment (asiegel-jt) [reply src/funk/fd_funk_txn.c:472]: done

**Comment by mmcgee-jump** [comment]
Just a reminder if we can change the commit message to reflect what's in here rather than "copying from private".

**Pull request merged**